File: Book/TestSQLLight/app/src/main/java/com/example/testsqllight/DatabaseHelper.java
package com.example.testsqllight;

import android.content.ContentValues;
import android.content.Context;
import android.database.Cursor;
import android.database.sqlite.SQLiteDatabase;
import android.database.sqlite.SQLiteOpenHelper;

import androidx.annotation.Nullable;

import java.util.ArrayList;

public class DatabaseHelper extends SQLiteOpenHelper {
    public static final String CUSTOMER_TABLE = "customer_table";
    public static final String CUSTOMER_NAME = "customer_name";
    public static final String CUSTOMER_AGE = "customer_age";
    public static final String VAL_CUSTOMER = "val_customer";

    public DatabaseHelper(@Nullable Context context) {
        super(context, "customer.db", null, 1);
    }

    @Override
    public void onCreate(SQLiteDatabase sqdb) {
        String sqlStatement= "CREATE TABLE " + CUSTOMER_TABLE + " (ID INTEGER PRIMARY KEY AUTOINCREMENT," +
                " " + CUSTOMER_NAME + " TEXT, " + CUSTOMER_AGE + " INTEGER, " +
                VAL_CUSTOMER + " BOOL)";
        sqdb.execSQL(sqlStatement);
    }
    public boolean addOne(Customer customer){
        SQLiteDatabase db=this.getWritableDatabase();
        ContentValues cv=new ContentValues();
        cv.put(CUSTOMER_NAME,customer.name);
        cv.put(CUSTOMER_AGE,customer.age);
        long insert = db.insert(CUSTOMER_TABLE, null, cv);
        if(insert>0){return true;
        }
        else{
            return false;
        }


    }

    public ArrayList<Customer> getList(){
        ArrayList<Customer> customerArrayList= new ArrayList<>();
        String sql_qury="SELECT * FROM "+CUSTOMER_TABLE;
        SQLiteDatabase db=this.getReadableDatabase();
        Cursor cursor = db.rawQuery(sql_qury, null);
        if(cursor.moveToNext()){
            do{
                int customer_id=cursor.getInt(0);
                String customer_name=cursor.getString(1);
                int customer_age=cursor.getInt(2);
                Customer customer=new Customer(customer_name,customer_age,customer_id);
                customerArrayList.add(customer);
            }while(cursor.moveToNext());
        }else{

        }

        return customerArrayList;
    }
    @Override
    public void onUpgrade(SQLiteDatabase sqLiteDatabase, int i, int i1) {

    }
}

File: practice22/app/src/main/java/com/example/practice2/DatabaseHelper.kt
package com.example.practice2

import android.database.sqlite.SQLiteOpenHelper
import android.content.ContentValues
import android.content.Context
import android.database.Cursor
import android.database.sqlite.SQLiteDatabase
import android.provider.BaseColumns

class DatabaseHelper(context: Context) :
    SQLiteOpenHelper(context, DatabaseContract.DATABASE_NAME, null, 1) {

    // Define SQL query to create the table
    private val SQL_CREATE_ENTRIES =
        "CREATE TABLE ${UserContract.TABLE_NAME} (" +
                "${UserContract.COLUMN_ID} INTEGER PRIMARY KEY," +
                "${UserContract.EMAIL} TEXT," +
                "${UserContract.COLUMN_PASSWORD} TEXT,"+
                 "${UserContract.first_name} TEXT,"+
                "${UserContract.last_name}TEXT,"+
                "${UserContract.Phone} TEXT)"

    override fun onCreate(db: SQLiteDatabase) {
        // Create the table
        db.execSQL(SQL_CREATE_ENTRIES)
    }

    override fun onUpgrade(db: SQLiteDatabase, oldVersion: Int, newVersion: Int) {
        // Handle database upgrades, if needed
    }

    private val dbHelper: DatabaseHelper = DatabaseHelper(context)
    private val db: SQLiteDatabase = dbHelper.writableDatabase

    fun insertUserName(name: String, pass:String): Long {
        val values = ContentValues().apply {
            put(UserContract.EMAIL, name)
            put(UserContract.COLUMN_PASSWORD,pass)
        }
        return db.insert(UserContract.TABLE_NAME, null, values)
    }

    fun getAllData(): List<String> {
        val dataList = ArrayList<String>()
        val projection = arrayOf(UserContract.TABLE_NAME)
        val selection = null
        val selectionArgs = null
        val sortOrder = null

        val cursor = readableDatabase.query(
            DatabaseContract.DATABASE_NAME,
            projection,
            selection,
            selectionArgs,
            null,
            null,
            sortOrder
        )

        while (cursor.moveToNext()) {

        }
        cursor.close()
        return dataList
    }

    fun deleteData(id: Long): Int {
        val selection = "${BaseColumns._ID} = ?"
        val selectionArgs = arrayOf(id.toString())
        return db.delete(UserContract.TABLE_NAME, selection, selectionArgs)
    }


}